ui/app/common/displaycontrols/pacsOrders/directives/pacsOrders.js

```diff
@@ -18,19 +18,63 @@ angular.module('bahmni.common.displaycontrol.pacsOrders')
                         visitUuid: $scope.visitUuid,
                         orderUuid: $scope.orderUuid
                     };
-                    return orderService.getOrders(params).then(function (response) {
-                        $scope.bahmniOrders = response.data;
-                        _.each($scope.bahmniOrders, function (order) {
-                            order.pacsImageUrl = ($scope.config.pacsImageUrl || "").replace('{{patientID}}', $scope.patient.identifier).replace('{{orderNumber}}', order.orderNumber);
-                        });
-                    });
+                    return orderService.getOrders(params);
                 };
-                var init = function () {
-                    return getOrders().then(function () {
+
+                var queryPacsStudies = function () {
+                    return pacsService.search($scope.patient.identifier);
+                };
+
+                var correlateWithStudies = function (radiologyOrders, radiologyStudies) {
+                    if (radiologyOrders) {
+                        radiologyOrders.forEach(function (ro) {
+                            ro.pacsImageUrl = ($scope.config.pacsImageUrl || "").replace('{{patientID}}', $scope.patient.identifier).replace('{{orderNumber}}', ro.orderNumber);
+                            if (radiologyStudies) {
+                                var matchingStudy = radiologyStudies.find((rs) => {
+                                    if (!rs.identifier) {
+                                        return false;
+                                    }
+                                    var matches = rs.identifier.filter((rsi) => {
```

Comment:
I feel like this could be extracted into its own function (parsing logic). And its corresponding test, to show the URLs that we support and how this works.

Reply:
done. added test spec

Reply:
Great! thanks Angshu!